src/microbe_stage/systems/FluidCurrentsSystem.cs

```diff
@@ -116,7 +160,55 @@ protected override void Update(float delta, in Entity entity)
         var pos = new Vector2(position.Position.X, position.Position.Z);
         var vel = VelocityAt(pos) * Constants.MAX_FORCE_APPLIED_BY_CURRENTS;
 
-        physicsControl.ImpulseToGive += new Vector3(vel.X, 0, vel.Y) * delta;
+        float effectStrength = entity.Get<CurrentAffected>().EffectStrength;
+
+        if (effectStrength == 0)
+        {
+            effectStrength = 1;
+        }
+        else if (effectStrength == -1)
+        {
+            effectStrength = 0;
+        }
+
+        physicsControl.ImpulseToGive += new Vector3(vel.X, 0, vel.Y) * delta * effectStrength;
         physicsControl.PhysicsApplied = false;
     }
+
+    private void TryGetNoiseImages()
+    {
+        var disturbancesX = noiseDisturbancesX.GetData();
+        var disturbancesY = noiseDisturbancesY.GetData();
+        var currentsX = noiseCurrentsX.GetData();
+        var currentsY = noiseCurrentsY.GetData();
+
+        if (disturbancesX == null || disturbancesY == null || currentsX == null || currentsY == null)
```

Comment:
I think it would be ever so slightly more efficient to check the values one by one and return early when one is missing.

Reply:
Will there be any difference? If the first statement is true the rest won't be evaluated because there are only || operators

Reply:
Yes because the variables are already retrieved before we start checking any of them. So by the time the if statement starts, the expensive work of fetching data from Godot C++ side is already done. So if the first one has already failed, we've tried to also fetch 3 useless pieces of data before we get to checking whether the first GetData call failed or not.